src/encoding_binding.cc

```diff
@@ -193,6 +194,31 @@ void BindingData::DecodeUTF8(const FunctionCallbackInfo<Value>& args) {
   args.GetReturnValue().Set(ret);
 }
 
+void BindingData::ToASCII(const v8::FunctionCallbackInfo<v8::Value>& args) {
+  Environment* env = Environment::GetCurrent(args);
+  CHECK_GE(args.Length(), 1);
+  CHECK(args[0]->IsString());
+
+  Utf8Value input(env->isolate(), args[0]);
+  auto out = ada::idna::to_ascii(input.ToStringView());
+  printf("to_ascii_returns '%s' for '%s'\n",
+         out.c_str(),
+         input.ToString().c_str());
```

Comment:
Nit: remove the printf. I assume this is leftover from debugging.

Reply:
That's a fun one!!!

Reply:
This is proof that I'm not using ChatGPT.